server/submission/api.ts

```diff
@@ -0,0 +1,54 @@
+import app, { wrap } from 'server/server';
+import { ForbiddenError } from 'server/utils/errors';
+
+import { getPermissions, canUpdate } from './permissions';
+import { createSubmission, updateSubmission, destroySubmission } from './queries';
+
+const getRequestIds = (req) => {
+	const user = req.user || {};
+	return {
+		userId: user.id,
+		communityId: req.body.communityId,
+		collectionId: req.body.collectionId,
+		submissionId: req.body.submissionId || null,
+	};
+};
+
+app.post(
+	'/api/submissions',
+	wrap(async (req, res) => {
+		const ids = getRequestIds(req);
+		const permissions = await getPermissions(ids);
+		if (!permissions.create) {
+			throw new ForbiddenError();
+		}
+		const newSubmission = await createSubmission(req.body);
```

Comment:
(Don't forget to pass the user ID into the `actorId` here)

Reply:
I actually took the opposite approach, and pulled all the activityItem stuff out, so that it can be added in its own PR. Seemed prudent.